db/migrate/20151024080849_collections_belongs_to_many_projects.rb

```diff
@@ -0,0 +1,8 @@
+class CollectionsBelongsToManyProjects < ActiveRecord::Migration
+  def change
+    add_column :collections, :project_ids, :integer, array: true, default: []
+    Collection.where.not(project_id: nil).update_all("project_ids = ARRAY[project_id]")
+    add_index  :collections, :project_ids, using: 'gin'
```

Comment:
yeah db index updates under heavy load (i.e. stargazing) would hit the db cpu really hard. This is on collections so will not have as many concurrent updates to deal with. If it's a problem we can drop it and figure out queries later.